File: MD.CMS.BusinessLogic.Core/DataAccess/Entities/FolderMetaDataField.cs
using MD.Tools.BaseDataAccess.Core.Entities;

namespace MD.CMS.BusinessLogic.Core.DataAccess.Entities
{
    public class FolderMetaDataField : BaseEntity<long>
    {
        #region Attributes
        private long _folderId;
        private bool _checked;
        private long _metaDataFieldId;
        private bool _isRequired;
        private string _name;
        #endregion

        #region Properties
        public long MetaDataFieldId
        {
            get { return _metaDataFieldId; }
            set { _metaDataFieldId = value; }
        }

        public bool IsRequired
        {
            get { return _isRequired; }
            set { _isRequired = value; }
        }

        public long FolderId
        {
            get { return _folderId; }
            set { _folderId = value; }
        }

        public bool Checked
        {
            get { return _checked; }
            set { _checked = value; }
        }

        public string Name { get => _name; set => _name = value; }
        #endregion

        #region Methods
        public FolderMetaDataField()
        {
        }

        public FolderMetaDataField(FolderMetaDataField obj) :
            base(obj)
        {
            this._checked = obj.Checked;
            this._isRequired = obj.IsRequired;
            this._folderId = obj.FolderId;
            this._metaDataFieldId = obj.MetaDataFieldId;
            this._name = obj.Name;
        }
        #endregion
    }
}

File: MD.CMS.Administration/MD.CMS.Administration.Core.Web/scripts/businessLogic/dataAccess/entities/folderMetaDataField.ts
/// <reference path="./base/iBaseEntity.ts" />
/// <reference path="./base/baseEntity.ts" />

namespace mdBusinessLogic.dataAccess.entities {
    export class folderMetaDataField extends base.BaseEntity implements base.IBaseEntity<folderMetaDataField> {
        public FolderId: number;
        public MetaDataFieldId: number;
        public IsRequired: boolean;
        public Checked: boolean;
        public Name: string;

        constructor(obj?: folderMetaDataField) {
            super(obj);
            this.FolderId = 0;
            this.MetaDataFieldId = 0;
            this.IsRequired = false;
            this.Checked = false;
            this.Name = '';
            if (obj != undefined && obj != null) {
                this.construct(obj);
            }
        }

        construct(data: any): void {
            super.construct(data);
            this.FolderId = this.getValue<number>(data, "FolderId", 0);
            this.MetaDataFieldId = this.getValue<number>(data, "MetaDataFieldId", 0);
            this.IsRequired = this.getValue<boolean>(data, "IsRequired", false);
            this.Checked = this.getValue<boolean>(data, "Checked", false);
            this.Name = this.getValue<string>(data, "Name", '');
        }

        public clone(): folderMetaDataField {
            return new folderMetaDataField(this);
        }

    }
}